### ISSUE TITLE
[Bug]: Cannot import any data from Freetube, Newpipe, Youtube, or Invidious.

### ISSUE THREAD

**Issue by Cyanide-1** [FreeTubeApp/FreeTube#3735]
### Guidelines

- [X] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [X] I have encountered this bug in the [official downloads of FreeTube](https://github.com/FreeTubeApp/FreeTube#official-downloads).
- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the bug report I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.
- [X] This issue contains only one bug.

### Describe the bug

How to achieve bug
1. go to data settings
2. click on "import subscriptions"
3. click on the desired file
4. no data is imported

### Expected Behavior

clicking on the file should import the data into freetube.

### Issue Labels

feature stopped working

### FreeTube Version

0.18.0 Beta

### Operating System Version

Void Linux x86-64 (Rolling Release)

### Installation Method

Flathub

### Primary API used

Invidious API

### Last Known Working FreeTube Version (If Any)

0.18.0 Beta

### Additional Information

basically i have tried importing data from every available source i could, none of it worked, not sure how to resolve this issue, i have use my freetube data from my last installation, as well as imported data using both invidious and newpipe, none of it seemed to register into freetube.

### Nightly Build

- [ ] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Hi @Cyanide-1 could u test this out using the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds). Or provide the file u would like to import so we can take a look.

**Comment by kwg** [comment]
I was able to duplicate the result with my subscription list from YT. After trial and error, I found a single entry to a channel that had been banned. I removed the channel and imported 477 entries without an issue. 

With the entry left in, the import would freeze and never provide an error or finish.

Confirmed with the current release build and nightly build 3106.

**Comment by Cyanide-1** [comment]
> Hi @Cyanide-1 could u test this out using the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds). Or provide the file u would like to import so we can take a look.

yeah sure give me a second

**Comment by Cyanide-1** [comment]
its not letting me upload the files for some reason

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@kwg @Cyanide-1 is it possible for u to provide the files u want to import into FT, so we can take a look at it. Please keep the banned channels in the file because we would like to troubleshoot whats going wrong.

**Comment by kwg** [comment]
[SubscriptionsError.csv](https://github.com/FreeTubeApp/FreeTube/files/12018412/SubscriptionsError.csv)
[SubscriptionsNoError.csv](https://github.com/FreeTubeApp/FreeTube/files/12018413/SubscriptionsNoError.csv)

**Comment by whowantsmybigdata** [comment]
Same problem on kubuntu 22.04.1 with FT-version 0.18.
can't import anything although its only like 15 subs and nothing is banned/removed.
unfortunately nightly-3112 doesn't solve it

**Comment by absidue** [comment]
Are you all using the Invidious API as your primary API?

**Comment by whowantsmybigdata** [comment]
tried both local and invidious...
btw subscribing to any new channel in FT works

**Comment by kwg** [comment]
Both files I posted have the same behavior in both APIs. My complete (large) list only works in local. That looks like a different issue.

**Comment by Cyanide-1** [comment]
> @kwg @Cyanide-1 is it possible for u to provide the files u want to import into FT, so we can take a look at it. Please keep the banned channels in the file because we would like to troubleshoot whats going wrong.

sorry for the wait, here is the link im trying to upload: https://jmp.sh/IDN9u99g

**Comment by Cyanide-1** [comment]
> Are you all using the Invidious API as your primary API?

been trying it with both invidious API and without, still doent work for some reason

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Data dump incoming (All tests were done on Local API):

Testing PR #2283:

- [x] I found that the listed `Test Subscriptions.db` in #2283 was successfully imported and i also saw the expected channels with errors section
- [x] csv listed in #2604 imported but with error `one or more subscriptions failed to import` which is expected
- [x] db listed in comment section of #3731 was successfully imported
- [x] SubscriptionsError.csv listed in comment section of #3735 was imported but with `one or more subscriptions failed to import` error which is expected
- [x] SubscriptionsNoError.csv listed in comment section of #3735 was imported but with `one or more subscriptions failed to import` error which is expected
- [ ] freetube-subscriptions.json listed in the comment section of #3735 did not import and i got error `TypeError: textDecode.forEach is not a function`

Testing v0.17.0 beta:

- [x] I found that the listed `Test Subscriptions.db` in #2283 was successfully imported and i also saw the expected channels with errors section
- [x] csv listed in #2604 imported but with error `one or more subscriptions failed to import` which is expected
- [x] db listed in comment section of #3731 was successfully imported
- [x] SubscriptionsError.csv listed in comment section of #3735 was imported but with `one or more subscriptions failed to import` error which is expected
- [x] SubscriptionsNoError.csv listed in comment section of #3735 was imported but with `one or more subscriptions failed to import` error which is expected
- [ ] freetube-subscriptions.json listed in the comment section of #3735 did not import and i got error `TypeError: textDecode.forEach is not a function`

Testing PR #2604:

- [x] I found that the listed `Test Subscriptions.db` in #2283 was successfully imported and i also saw the expected channels with errors section
- [x] csv listed in #2604 imported but with error `one or more subscriptions failed to import` which is expected (i thought this error would be fixed by this PR but i could be wrong)
- [x] db listed in comment section of #3731 was successfully imported
- [x] SubscriptionsError.csv listed in comment section of #3735 was imported but now i get another error `failed to import kraz3yivan` 
- [x] SubscriptionsNoError.csv listed in comment section of #3735 was imported successfully
- [] freetube-subscriptions.json listed in the comment section of #3735 did not import and got no error

Testing v0.18.0 beta:

- [x] I found that the listed `Test Subscriptions.db` in #2283 was successfully imported and i also saw the expected channels with errors section
- [x] csv listed in #2604 imported but with error `one or more subscriptions failed to import` which is expected and also got an error in console `failed to import Deleted Channel`
- [x] db listed in comment section of #3731 imported successfully
- [x] SubscriptionsError.csv listed in comment section of #3735 was imported but now i get another error `failed to import kraz3yivan` 
- [x] SubscriptionsNoError.csv listed in comment section of #3735 was imported successfully
- [] freetube-subscriptions.json listed in the comment section of #3735 did not import and got no error

Testing latest dev:

- [x] I found that the listed `Test Subscriptions.db` in #2283 imported successfully and i also saw the expected channels with errors section
- [] csv listed in #2604 did not import at all and i got an error `TypeError: Cannot read properties of undefined (reading 'author')`
- [x] db listed in comment section of #3731 imported successfully
- [] SubscriptionsError.csv listed in comment section of #3735 did not import at all and i got an error `TypeError: Cannot read properties of undefined (reading 'author')`
- [x] SubscriptionsNoError.csv listed in comment section of #3735 imported successfully 
- [] freetube-subscriptions.json listed in the comment section of #3735 did not import and got no error
